app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -1103,4 +1111,51 @@ public function getOrderByQuoteId($quoteId) {
                 ->addFieldToFilter('quote_id', $quoteId)
                 ->getFirstItem();
     }
-}
\ No newline at end of file
+
+    /**
+     * Gets the immutable quote id stored in the Bolt transaction.  This is backwards
+     * compatible with older versions of the plugin and is suitable for transition
+     * installations.
+     *
+     * @param object $transaction  The Bolt transaction as a php object
+     *
+     * @return string  The immutable quote id
+     */
+    public function getImmutableQuoteIdFromTransaction( $transaction ) {
+        if (strpos($transaction->order->cart->display_id, '|')) {
+            return explode("|", $transaction->order->cart->display_id)[1];
+        } else {
+            /////////////////////////////////////////////////////////////////
+            // here we deal with older versus newer version
```

Comment:
Newer version is handled by above `return explode("|", $transaction->order->cart->display_id)[1];`, isn't it?

Reply:
@daisy1754 correct.

Reply:
@daisy1754 When you push this code into production on pre-existing merchants, there will be several orders created that use the old format.  If there is no support for the legacy format, all webhooks for these orders would fail.  A good example would be payments that are in review that will update themselves or issuing a refund or canceling an order from the merchant dashboard.   This was pointed out to me by @danac-gs who also expressed a need for this addition and is the author of this excellent idea.

So, I will change the wording in this comment to 
`here we deal with the legacy hook format for backward compatibility`, which definitely expresses this more clearly.

I also already give a brief explanation of this in the methods comments:

https://github.com/BoltApp/bolt-magento1/blob/2cf827d0485710707830c478511468cf6df268ce/app/code/community/Bolt/Boltpay/Helper/Api.php#L1115-L1124